lang/py/avro/datafile.py

```diff
@@ -80,10 +81,65 @@ def __init__(self, fail_msg):
 # Write Path
 #
 
-class DataFileWriter(object):
-  @staticmethod
-  def generate_sync_marker():
-    return generate_sixteen_random_bytes()
+class _DataFile(object):
+  """Mixin for methods common to both reading and writing."""
+
+  block_count = 0
+  _meta = None
```

Comment:
Should these be class variables?  There's a `self.block_count` in one of the subclasses as well.

Reply:
You mean versus instance variables? It doesn't really make a difference for most usage.

```python
>>> class Foo(object):
...   bar = 123
...   def m(self):
...     self.bar = 'abc'
... 
>>> f = Foo()
>>> f.bar
123
>>> g = Foo()
>>> g.bar
123
>>> f.m()
>>> f.bar
'abc'
>>> g.bar
123
```

Here, setting `block_count` as a constant class variable just makes it a default for the instance. Things do get gnarly if you use a mutable type here, but mutable default types are to be avoided in any case.